File: lib/kwartz/binding/pierubis.rb
###
### $Rev$
### $Release$
### $Copyright$
###

require 'kwartz/converter'
require 'kwartz/translator'
require 'kwartz/binding/ruby'



module Kwartz



  class PIErubisHandler < RubyHandler
  end
  Handler.register_class('pierubis', PIErubisHandler)



  ##
  ## translator for Erubis
  ##
  class PIErubisTranslator < BaseTranslator


    PIERUBIS_EMBED_PATTERNS = [
      '<?rb ', ' ?>',      # statement
      '$!{',   '}',        # expression
      '${',    '}',        # escaped expression
    ]


    def initialize(properties={})
      super(PIERUBIS_EMBED_PATTERNS, properties)
      #@escape = true if @escape == nil
    end


    def translate_native_expr(expr)
      assert unless expr.is_a?(NativeExpression)
      if expr.code.include?(?})
        @expr_l, @expr_r, @escape_l, @escape_r = '<%=', '%>', '<%==', '%>'
      else
        @expr_l, @expr_r, @escape_l, @escape_r = '$!{', '}', '${', '}'
      end
      super(expr)
    end


  end
  Translator.register_class('pierubis', PIErubisTranslator)



end #module

File: Kwartz/Binding/Pierubis.php
<?php
// vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4:

// $Rev$
// $Release$
// $Copyright$

require_once 'Kwartz/Converter.php';
require_once 'Kwartz/Translator.php';
require_once 'Kwartz/Binding/Ruby.php';


/**
 *  directive handler for PIEruby
 */
class KwartzPierubisHandler extends KwartzRubyHandler {
}



/**
 *  translator for eRuby
 */
class KwartzPierubisTranslator extends KwartzBaseTranslator {


    function __construct($properties=array()) {
        $marks = array(
            '<?rb ',  ' ?>',    // statement
            '@!{',    '}@',     // expression
            '@{',     '}@',     // escaped expression
            );
        parent::__construct($marks, $properties);
        //if ($this->escape === null)
        //    $this->escape = true;
    }


}


?>
